tests/model_registry/model_catalog/test_model_search.py

```diff
@@ -294,3 +296,228 @@ def test_multiple_artifact_type_filtering(
+        )
+        assert is_valid, f"API search results do not match database query for '{nonexistent_term}': {errors}"
+
+        models = response.get("items", [])
+        assert len(models) == 0, f"Expected no results for '{nonexistent_term}', got {len(models)} models"
+
+    @pytest.mark.parametrize("search_term", ["", None])
+    def test_q_parameter_empty_query(
+        self: Self,
+        search_term,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+        model_registry_namespace: str,
+    ):
+        """Test behavior with empty or None q parameter using database validation"""
+        LOGGER.info(f"Testing empty query: {repr(search_term)}")
+
+        response = get_models_from_catalog_api(
+            model_catalog_rest_url=model_catalog_rest_url,
+            model_registry_rest_headers=model_registry_rest_headers,
+            q=search_term,
+        )
+
+        # Only validate against database if search_term is not None/empty
+        # Empty/None queries might have different behavior that doesn't map to our search logic
+        if search_term:
+            is_valid, errors = validate_search_results_against_database(
+                api_response=response,
+                search_term=search_term,
+                namespace=model_registry_namespace,
+            )
+            assert is_valid, f"API search results do not match database query for '{search_term}': {errors}"
+
+        models = response.get("items", [])
+        LOGGER.info(f"Empty/None query returned {len(models)} models")
+
+    def test_q_parameter_with_source_label_filter(
+        self: Self,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+        model_registry_namespace: str,
+    ):
+        """Test q parameter combined with source_label filtering using database validation"""
+        search_term = "granite"
+        source_label = REDHAT_AI_FILTER
+
+        LOGGER.info(f"Testing combined search: q='{search_term}' with sourceLabel='{source_label}'")
+
+        response = get_models_from_catalog_api(
+            model_catalog_rest_url=model_catalog_rest_url,
+            model_registry_rest_headers=model_registry_rest_headers,
+            q=search_term,
+            source_label=source_label,
+        )
+
+        models = response.get("items", [])
+        LOGGER.info(f"Combined filter returned {len(models)} models")
+
+        # Validate combined results against database query
+        is_valid_combined, errors_combined = validate_search_results_against_database(
+            api_response=response,
+            search_term=search_term,
+            source_label=source_label,
+            namespace=model_registry_namespace,
+        )
+        assert is_valid_combined, (
+            f"API combined search+source_label results do not match database query for '{search_term}' with '{source_label}': {errors_combined}"  # noqa: E501
```

Comment:
for this search term, you already validated the db once, as a part of test_q_parameter_basic_search(). This is not needed.